crates/flashblocks-archiver/src/types.rs

```diff
@@ -0,0 +1,122 @@
+use alloy_primitives::{map::foldhash::HashMap, Address, B256, U256};
+use alloy_rpc_types_engine::PayloadId;
+use chrono::{DateTime, Utc};
+use reth_optimism_primitives::OpReceipt;
+use rollup_boost::{ExecutionPayloadBaseV1, ExecutionPayloadFlashblockDeltaV1};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FlashblockMessage {
+    pub payload_id: PayloadId,
+    pub index: u64,
+    pub base: Option<ExecutionPayloadBaseV1>,
+    pub diff: ExecutionPayloadFlashblockDeltaV1,
+    pub metadata: Metadata,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default)]
+pub struct Metadata {
+    pub receipts: HashMap<B256, OpReceipt>,
+    pub new_account_balances: HashMap<Address, U256>,
+    pub block_number: u64,
+}
```

Comment:
I wonder if it's worth exporting/importing these types from the flashblocks-rpc crate (along with the deserialization logic)